examples/contrib/oed/eig_estimation_benchmarking.py

```diff
@@ -7,148 +9,383 @@
+      (donsker_varadhan_eig,
+       [400, 400, GuideDV(basic_2p_ba_guide(10)),
+        optim.Adam({"lr": 0.05}), False, None, 500]),
+      (ba_eig_lm,
+       [20, 400, basic_2p_ba_guide(10), optim.Adam({"lr": 0.05}),
+        False, None, 500])
+      ]),
+    ("A/B test linear model known covariance (different sds)",
+     basic_2p_linear_model_sds_10_0pt1, AB_test_11d_10n_2p, "y", "w",
+     [(linear_model_ground_truth, []),
+      (naive_rainforth_eig, [2000, 2000]),
+      (vi_eig_lm,
+       [{"guide": basic_2p_guide, "optim": optim.Adam({"lr": 0.05}), "loss": elbo,
+         "num_steps": 1000}, {"num_samples": 1}]),
+      (donsker_varadhan_eig,
+       [400, 400, GuideDV(basic_2p_ba_guide(11)),
+        optim.Adam({"lr": 0.05}), False, None, 500]),
+      (ba_eig_lm,
+       [20, 400, basic_2p_ba_guide(11), optim.Adam({"lr": 0.05}),
+        False, None, 500])
+      ]),
 ])
-def test_eig_and_plot(arglist):
+def test_eig_and_plot(title, model, design, observation_label, target_label, arglist):
     """
     Runs a group of EIG estimation tests and plots the estimates on a single set
     of axes. Typically, each test within one `arglist` should estimate the same quantity.
     This is repeated for each `arglist`.
     """
-    pyro.set_rng_seed(42)
     ys = []
-    for design_tensor, estimator, args in arglist:
-        ys.append(time_eig(design_tensor, estimator, *args))
+    names = []
+    elapseds = []
+    print(title)
+    for estimator, args in arglist:
+        y, elapsed = time_eig(estimator, model, design, observation_label, target_label, args)
+        ys.append(y)
+        elapseds.append(elapsed)
+        names.append(estimator.name)
 
     if PLOT:
         import matplotlib.pyplot as plt
-        plt.figure(figsize=(12, 8))
+        plt.figure(figsize=(10, 5))
         for y in ys:
             plt.plot(y.detach().numpy(), linestyle='None', marker='o', markersize=10)
+        plt.title(title)
+        plt.legend(names)
+        plt.xlabel("Design")
+        plt.ylabel("EIG estimate")
         plt.show()
 
 
-def time_eig(design_tensor, estimator, *args):
+def time_eig(estimator, model, design, observation_label, target_label, args):
     pyro.clear_param_store()
 
     t = time.time()
-    y = estimator(design_tensor, *args)
+    y = estimator(model, design, observation_label, target_label, *args)
     elapsed = time.time() - t
 
-    print(y)
-    print(elapsed)
-    return y
+    print(estimator.__name__)
+    print('estimate', y)
+    print('elapsed', elapsed)
+    return y, elapsed
 
 
-@pytest.mark.parametrize("design,w_sds,dv_params", [
-     (X_small, torch.tensor([10., 2.5]), {"n_iter": 2000, "n_samples": 2000, "lr": 0.0005, "T": T_neural(2, 2),
-                                          "final_n_samples": 10000}),
+@pytest.mark.parametrize("title,model,design,observation_label,target_label,est1,est2,kwargs1,kwargs2", [
+    ("Barber-Agakov on difficult sigmoid",
```

Comment:
Move to list of tuples as in e.g. [HMC tests](https://github.com/uber/pyro/blob/dev/tests/infer/mcmc/test_hmc.py#L60)